Tests fail when upgrading glib from 2.22 to 2.24 if sys-devel/gdb is installed
because gdb is run on .libs/assert-msg-test before LD_LIBRARY_PATH is set. This
causes gdb to use the system-wide glib instead, and fail on the test.

This patch exports LD_LIBRARY_PATH before running gdb

https://bugzilla.gnome.org/621368

---
--- tests/run-assert-msg-test.sh
+++ tests/run-assert-msg-test.sh
@@ -34,6 +34,7 @@ if [ -e ".libs/lt-$msg_test" ]; then
        msg_test="lt-$msg_test"
 fi
 echo_v "Running gdb on assert-msg-test"
+export LD_LIBRARY_PATH="`dirname $PWD`/glib/.libs:$LD_LIBRARY_PATH"
 OUT=$(gdb --batch --ex run --ex "print (char*) __glib_assert_msg" .libs/$msg_test 2> $error_out) || \
   fail "failed to run gdb"
 
